ui/app/src/components/FormsEngine/controls/Text.tsx

```diff
@@ -15,26 +15,74 @@
  */
 
 import OutlinedInput, { OutlinedInputProps } from '@mui/material/OutlinedInput';
-import React, { useId } from 'react';
+import React, { useEffect, useId, useMemo, useState } from 'react';
 import { FormsEngineField } from '../components/FormsEngineField';
 import { ControlProps } from '../types';
+import { escapeXml, unescapeXml } from '../../../utils/xml';
+import { isFieldRequired } from '../lib/validators';
+import { nnou, nou } from '../../../utils/object';
 
 export interface TextProps extends ControlProps {
 	value: string;
 }
 
 export function Text(props: TextProps) {
-	const { field, value, setValue, readonly, autoFocus } = props;
+	const { field, value: valueProp, setValue, readonly: formReadonly, autoFocus } = props;
 	const htmlId = useId();
-	const maxLength = field.validations.maxLength?.value;
-	const handleChange: OutlinedInputProps['onChange'] = (e) => setValue(e.currentTarget.value);
+
+	// region field properties/validations
+	const maxLength = field.validations?.maxLength?.value;
+	const readonly = formReadonly || (field.properties?.readonly?.value as boolean);
+	const escapeContent = (field.properties?.escapeContent?.value as boolean) ?? false;
+	const pattern = field.validations?.pattern?.value as string;
+	const defaultValue = field.defaultValue as string;
+	// endregion
+	const isRequired = isFieldRequired(field);
+	const rawValue = nnou(valueProp) ? valueProp : (defaultValue ?? '');
+	const [patternError, setPatternError] = useState(false);
+
+	const value = useMemo(() => (escapeContent ? unescapeXml(rawValue) : rawValue), [rawValue, escapeContent]);
```

Comment:
This seems to break the chain. The value used should be the one coming from the engine. This seems disconnected.